### PR TITLE
Mtl h dgpu

### PR DESCRIPTION
**Pull request by mkopec**

### CODE REVIEW

File: src/mainboard/google/brox/acpi/power.asl [hunk 1]
```diff
@@ -199,39 +199,13 @@ Method (PGON, 0, Serialized)
 	/* Assert PERST# */
 	CTXS (GPIO_GPU_PERST_L)
 
-	/* Ramp up 1.2V rail on boards with support */
```
Comment (mkopec) [inline src/mainboard/google/brox/acpi/power.asl:202]: whoopsie

**Comment by miczyg1** [comment]
@mkopec something wrong happened, because the PR contains changes from iGPU variant

**Comment by mkopec** [comment]
@miczyg1 rebased, should be okay now

**Review by miczyg1 — COMMENTED** [review]
Not sure how useful these comments will be. Without a spec it is hard to say what should be done and how. There is a lot of other things in original firmware ACPI. Leaving it a sa comment rather than change request,

File: src/mainboard/clevo/mtl-h/acpi/dgpu/gps.asl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+				Return (GPSR)
+			}
+		}
+		Case (GPS_FUNC_PSHARESTATUS)
+		{
+			Return (ITOB(0))
+		}
+		Case (GPS_FUNC_PSHAREPARAMS)
+		{
+			CreateField (Arg1, 0, 4, QUTY)	 /* Query type */
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/gps.asl:62]: Original firmware has all functions in the GPU's _DSM method, here we have a separate method for each function and call GPS(Arg2, Arg3) in the _DSM

File: src/mainboard/clevo/mtl-h/acpi/dgpu/nvjt.asl [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+				(0 << 3) |		/* PPR: Panel Power Rail */
+				(0 << 5) |		/* SRPR: Self-Refresh Controller Power Rail */
+				(0 << 6) |		/* FBPR: FB Power Rail */
+				(0 << 8) |		/* GPR: GPU Power Rail */
+				(0 << 10) |		/* GCR: GC6 ROM */
+				(1 << 11) |		/* PTH: No SMI Handler */
+				(0 << 12) |		/* NOT: Supports Notify on GC6 State done */
+				(1 << 13) |		/* MHYB: MS Hybrid Support (deferred GC6) */
+				(0 << 14) |		/* RPC: Root Port Control */
+				(0 << 15) |		/* GC6 Version (GC6-E) */
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/nvjt.asl]: Changed to 0x2 as in original fw

File: src/mainboard/clevo/mtl-h/acpi/dgpu/nvjt.asl [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+{
+	Switch (ToInteger(Arg0))
+	{
+		Case (JT_FUNC_SUPPORT)
+		{
+			Return (ITOB(
+				(1 << JT_FUNC_SUPPORT) |
+				(1 << JT_FUNC_CAPS) |
+				(1 << JT_FUNC_POWERCONTROL) |
+				(1 << JT_FUNC_PLATPOLICY)))
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/nvjt.asl]: yep, removed

File: src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+			Printf("NVPCF: Get static config tables")
+			Return (Buffer () {
+				/* System Device Table Header (v2.0) */
+				0x20, 0x03, 0x01,
+
+				/* System Device Table Entries */
+				0x00,			/* [3:0] CPU type (0=Intel, 1=AMD),
+							   [7:4] GPU type (0=Nvidia) */
+
+				/* System Controller Table Header (v2.2), 1 controller entry */
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl:43]: Changed to 2.4 in https://github.com/Dasharo/coreboot/pull/548/commits/950f78bdbfaa995ac71ff9a140d9469c6d9f253f

File: src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+		Case (NVPCF_FUNC_UPDATE_DYNAMIC_PARAMS)
+		{
+			Printf("NVPCF: Update dynamic params")
+
+			CreateField(Arg1, 0x28, 2, ICMD)
+			Printf("NVPCF: ICMD = %o", ICMD)
+
+			Local0 = Buffer (0x31) {
+				/* Dynamic Params Table Header (1 controller entry, 0x1c bytes) */
+				0x22, 0x05, 0x10, 0x1c, 0x01 }
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl]: Changed to 2.4 in https://github.com/Dasharo/coreboot/pull/548/commits/950f78bdbfaa995ac71ff9a140d9469c6d9f253f

File: src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+				0x01,			/* [3:0] Controller class
+							      0=Disabled, 1=Dynamic Boost,
+							      2=CTGP-only.
+							   [7:4] Reserved. Set to 0. */
+				/* Controller #1 Params */
+							/* Class = Dynamic Boost
+							   [0:0] DC support
+							      0=Not supported, 1=Supported
+							   [31:1] Reserved. Set to 0. */
+				0x00, 0x00, 0x00, 0x00,
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl]: Changed to 0x1 in https://github.com/Dasharo/coreboot/pull/548/commits/950f78bdbfaa995ac71ff9a140d9469c6d9f253f

File: src/mainboard/clevo/mtl-h/acpi/dgpu/gps.asl [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+#define QUERY_GET_SUPPORTED_FIELDS	1
+#define QUERY_GET_CURRENT_LIMITS	2
+
+#define EC_D_NOTIFY_MASK	0x7
+
+/* GPS return Package */
+Name (GPSP, Buffer (0x28) {0x0})
+CreateDWordField (GPSP, 0, RETN)
+CreateDWordField (GPSP, 4, VRV1)
+CreateDWordField (GPSP, 8, TGPU)
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/gps.asl:16]: Yes, but it doesn't actually populate most of them. Tested by calling the DSM by acpidbg, we should now be returning the same exact values

File: src/mainboard/clevo/mtl-h/acpi/dgpu/nvop.asl [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+			}
+			Else
+			{
+				Local0 = 0
+			}
+
+			Return (ITOB (
+				(1 << 0) |	/* Optimus Enabled */
+				(Local0 << 3) |	/* Current GPU Control Status */
+				(0 << 6) |	/* Shared Discrete GPU Hotplug Capabilities */
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/nvop.asl]: yes, fixed in https://github.com/Dasharo/coreboot/pull/548/commits/554b771a19486fb4979cdfc84fe57ac18600f291

File: src/mainboard/clevo/mtl-h/acpi/dgpu/gpu_top.asl [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+				Default { Return (D5_GPU) }
+			}
+		}
+
+		/* Current D Notify Value, defaults to D1
+		* Arg0 == Shared value
+		* Arg1 == force notification if no change (0 or 1)
+		*/
+		Name (CDNV, D1_EC)
+		Method (DNOT, 2, Serialized)
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/mtl-h/acpi/dgpu/gpu_top.asl:47]: It's now used: https://github.com/Dasharo/coreboot/pull/548/commits/bab51b072f04e36df4db7cbeca4f09e4723dd1fd

File: .github/workflows/build.yml [hunk 11]
```diff
@@ -11,37 +11,16 @@ on:
       - '*'
 
 jobs:
-  build_novacustom_ec:
```
Comment (miczyg1) [inline .github/workflows/build.yml]: @mkopec I think we can still build EC regardless if EC sync is used or not. The "Obtain EC" step is removed and the dependency too, so it should be fine.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 11 above)
Comment (mkopec) [reply .github/workflows/build.yml]: Right, commit amended: https://github.com/Dasharo/coreboot/commit/c2b15730096a54a4518709d9390e66fa8d58acea

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**